File: asset_leasing/events/customer.py
"""Customer validate handler - P1-08."""

import frappe
from frappe import _
from frappe.utils import flt


def validate(doc, method=None):
	# VAL-P1-08
	if doc.get("al_hire_blocked") and not (doc.get("al_hire_block_reason") or "").strip():
		frappe.throw(_("Record a <b>Reason for Block</b> when blocking a customer for hire."))

	if flt(doc.get("al_deposit_held")) < 0:
		frappe.throw(_("Security Deposit Held cannot be negative."))

File: asset_leasing/public/js/customer.js
// Customer - surface a hire block prominently.

frappe.ui.form.on('Customer', {
	refresh(frm) {
		if (frm.is_new() || !frm.doc.al_hire_blocked) return;
		frm.dashboard.set_headline(
			__('Blocked for hire — {0}', [frm.doc.al_hire_block_reason || __('no reason recorded')]),
			'red'
		);
	},
});
